10up/retro-winamp-block
# Feature/setup cypress for e2e testing

### Description of the Change

This PR adds Cypress for e2e testing, configured and with 2 tests set-up:

1. block.test - the test adds a block to the page, uploads a test mp3, assigns it to the block, saves the post and verifies the block exists on the newly created page
2. admin.test - the test logs in and deactivates and activates the plugin


Closes #47 

We also bump WordPress minimum support to 6.1

Closes #88

### How to test the Change

1. Pull changes
2. Run `npm install`
3. Build assets: `npm run build`
4. Start local environmation: `npm run env:start`
5. Open Cypress for testing: `npm run cypress:open`
6. Use Cypress GUI to run tests and verify they are working

### Changelog Entry

> Added - e2e testing with Cypress
> Tweak - Bump WordPress requires at least 6.1


### Credits
Props @jamespstyles for help getting Cypress set-up


### Checklist:
